plugins/file/plugin.go

```diff
@@ -125,24 +209,137 @@ func setupFile6(args ...string) (handler.Handler6, error) {
+		}
+		filename := args[0]
+		if filename == "" {
+			return nil, nil, errors.New("plugins/file: got empty file name")
+		}
+		records, err := LoadDHCPv6Records(filename)
+		if err != nil {
+			return nil, nil, fmt.Errorf("plugins/file: failed to load DHCPv6 records: %v", err)
+		}
+		log.Printf("plugins/file: loaded %d leases from %s", len(records), filename)
+		StaticRecords = records
+	} else {
+		if len(args) < 3 {
+			return nil, nil, errors.New("plugins/file: need a file name, server IP, netmask and a lease time")
+		}
+		filename = args[0]
+		if filename == "" {
+			return nil, nil, errors.New("plugins/file: got empty file name")
+		}
+		ip, network, err := net.ParseCIDR(args[1])
+		if err != nil {
+			return Handler6, Handler4, errors.New("plugins/file: expected an IPv4 address, got: " + args[1])
+		}
+		serverIP = ip
+
+		netmask = network.Mask
+
+		leaseTime, err := strconv.ParseUint(args[2], 10, 32)
+		if err != nil {
+			return Handler6, Handler4, errors.New("plugins/file: expected an uint32, got: " + args[2])
+		}
+		LeaseTime = uint32(leaseTime)
+		records, err := LoadDHCPv4Records(filename)
+		if err != nil {
+			return nil, nil, fmt.Errorf("plugins/file: failed to load DHCPv4 records: %v", err)
+		}
+		DHCPv4Records = records
+		log.Printf("plugins/file: loaded %d leases from %s", len(DHCPv4Records), filename)
 	}
-	filename := args[0]
-	if filename == "" {
-		return nil, nil, errors.New("plugins/file: got empty file name")
+
+	return Handler6, Handler4, nil
+}
+func createIP(serverIP net.IP, netmask net.IPMask) (Record, error) {
+
+	rand.Seed(time.Now().Unix())
+	ipserver := serverIP.To4()
+
+	ip := net.IPv4(random(1, 254), random(1, 254), random(1, 254), random(1, 254)).To4()
+	for i := 0; i < 4; i++ {
+		ip[i] = (ip[i] & (netmask[i] ^ 255)) | (ipserver[i] & netmask[i])
 	}
-	records, err := LoadDHCPv6Records(filename)
+	taken := checkIfTaken(net.IPv4(ip[0], ip[1], ip[2], ip[3]))
+	for taken {
+		ipInt := binary.BigEndian.Uint32(ip)
+		ipInt++
+		nextIP := make([]byte, 4)
+		binary.BigEndian.PutUint32(ip, ipInt)
+		for i := 0; i < 4; i++ {
+			nextIP[i] = (ip[i] & (netmask[i] ^ 255)) | (ipserver[i] & netmask[i])
+		}
+		if nextIP[0] != ip[0] || nextIP[1] != ip[1] || nextIP[2] != ip[2] || nextIP[3] != ip[3] {
+			break
+		}
+		ip = nextIP
+		taken = checkIfTaken(net.IPv4(ip[0], ip[1], ip[2], ip[3]))
+	}
+	for taken {
+		ipInt := binary.BigEndian.Uint32(ip)
+		ipInt--
+		nextIP := make([]byte, 4)
+		binary.BigEndian.PutUint32(ip, ipInt)
+		for i := 0; i < 4; i++ {
+			nextIP[i] = (ip[i] & (netmask[i] ^ 255)) | (ipserver[i] & netmask[i])
+		}
+		if nextIP[0] != ip[0] || nextIP[1] != ip[1] || nextIP[2] != ip[2] || nextIP[3] != ip[3] {
+			return Record{}, errors.New("plugins/file: no new IP addresses available")
+		}
+		ip = nextIP
+		taken = checkIfTaken(net.IPv4(ip[0], ip[1], ip[2], ip[3]))
+
+	}
+	return Record{IP: net.IPv4(ip[0], ip[1], ip[2], ip[3]), leaseTime: LeaseTime, leased: time.Now().Unix()}, nil
```

Comment:
Why `Record{IP: net.IPv4(ip[0], ip[1], ip[2], ip[3]), leaseTime: LeaseTime, leased: time.Now().Unix()}, nil` instead of `Record{IP: ip), leaseTime: LeaseTime, leased: time.Now().Unix()}, nil`?